src/main/java/com/amazon/ion/impl/macro/SystemMacro.kt

```diff
@@ -2,15 +2,27 @@
 // SPDX-License-Identifier: Apache-2.0
 package com.amazon.ion.impl.macro
 
+import com.amazon.ion.impl.*
+import com.amazon.ion.impl.SystemSymbols_1_1.*
+import com.amazon.ion.impl.macro.ExpressionBuilderDsl.Companion.templateBody
 import com.amazon.ion.impl.macro.ParameterFactory.exactlyOneFlexInt
 import com.amazon.ion.impl.macro.ParameterFactory.exactlyOneTagged
 import com.amazon.ion.impl.macro.ParameterFactory.oneToManyTagged
+import com.amazon.ion.impl.macro.ParameterFactory.zeroOrOneTagged
 import com.amazon.ion.impl.macro.ParameterFactory.zeroToManyTagged
 
 /**
  * Macros that are built in, rather than being defined by a template.
  */
-enum class SystemMacro(val id: Byte, val macroName: String, override val signature: List<Macro.Parameter>) : Macro {
+enum class SystemMacro(val id: Byte, val macroName: String, override val signature: List<Macro.Parameter>, override val body: List<Expression.TemplateBodyExpression>? = null) : Macro {
+    // Technically not system macros, but special forms. However, it's easier to model them as if they are macros in TDL.
+    // We give them an ID of -1 to distinguish that they are not addressable outside TDL.
+    IfNone(-1, "if_none", listOf(zeroToManyTagged("stream"), zeroToManyTagged("true_branch"), zeroToManyTagged("false_branch"))),
+    IfSome(-1, "if_some", listOf(zeroToManyTagged("stream"), zeroToManyTagged("true_branch"), zeroToManyTagged("false_branch"))),
+    IfSingle(-1, "if_single", listOf(zeroToManyTagged("stream"), zeroToManyTagged("true_branch"), zeroToManyTagged("false_branch"))),
+    IfMulti(-1, "if_multi", listOf(zeroToManyTagged("stream"), zeroToManyTagged("true_branch"), zeroToManyTagged("false_branch"))),
```

Comment:
🗺️ These were moved to the beginning of the enum because `use` needs to reference `if_none`, and the constructor args for an enum entry can only use _back_ references to other entries of the same enum.